src/main/java/org/prebid/server/bidder/huaweiads/HuaweiAdsBidder.java

```diff
@@ -0,0 +1,1267 @@
+                + "<ClickThrough><![CDATA[" + clickUrl + "]]></ClickThrough>"
+                + dspClickTracking2Str
+                + "</VideoClicks>"
+                + "<MediaFiles>"
+                + "<MediaFile delivery=\"progressive\" type=\"" + mime + "\" width=\"" + adWidth + "\" "
+                + "height=\"" + adHeight + "\" scalable=\"true\" maintainAspectRatio=\"true\"> "
+                + "<![CDATA[" + resourceUrl + "]]>"
+                + "</MediaFile>"
+                + "</MediaFiles>"
+                + "</Linear>"
+                + "</Creative>" + rewardedVideoPart
+                + "</Creatives>"
+                + "</InLine></Ad></VAST>";
+
+        return bidBuilder.adm(adm).build();
+    }
+
+    private void handleEvent(StringBuilder dspImpTracking2Str,
+                             StringBuilder dspClickTracking2Str,
+                             StringBuilder errorTracking2Str,
+                             StringBuilder trackingEvents,
+                             Monitor monitor) {
+        String event = "";
+        switch (monitor.getEventType()) {
+            case "vastError":
+                errorTracking2Str.append(
+                        getVastImpClickErrorTrackingUrls(monitor.getUrl(), "vastError"));
+                break;
+            case "imp":
+                dspImpTracking2Str.append(
+                        getVastImpClickErrorTrackingUrls(monitor.getUrl(), "imp"));
+                break;
+            case "click":
+                dspClickTracking2Str.append(
+                        getVastImpClickErrorTrackingUrls(monitor.getUrl(), "click"));
+                break;
+            case "userclose":
+                event = "skip&closeLinear";
+                break;
+            case "playStart":
+                event = "start";
+                break;
+            case "playEnd":
+                event = "complete";
+                break;
+            case "playResume":
+                event = "resume";
+                break;
+            case "playPause":
+                event = "pause";
+                break;
+            case "soundClickOff":
+                event = "mute";
+                break;
+            case "soundClickOn":
+                event = "unmute";
+                break;
+            default:
+                break;
+        }
+
+        if (!event.isEmpty() && !event.equals("skip&closeLinear")) {
+            trackingEvents.append(getVastEventTrackingUrls(monitor.getUrl(), event));
+        } else if (!event.isEmpty()) {
+            trackingEvents.append(getVastEventTrackingUrls(monitor.getUrl(), "skip&closeLinear"));
+        }
+    }
+
+    private String getDuration(Long duration) {
+        final Duration dur = Duration.ofMillis(duration);
+        final LocalTime time = LocalTime.MIDNIGHT.plus(dur);
+        final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("HH:mm:ss.SSS");
+        return time.format(formatter);
+    }
```

Comment:
methods like this should not start with `get`. You already have duration as a parameter, so you are not **getting** it, you are **transforming** it (or **converting** if you will).